app.ts

```diff
@@ -0,0 +1,137 @@
+let recordCount: number, startIndex: number, endIndex: number;
+let columnHeaders: any[];
+let resizeTimer: number , rowsPerView : number;
+
+window.onload = function(){
+  getRecordCount();
+  calRowPerView();
+}
+
+// calculating how many records to display based on widow size
+function calRowPerView() {
+  rowsPerView = Math.floor((window.innerHeight - 245) / 34);
+  
+  if (rowsPerView < 0) {
+    alert("Window screen to small to display records , Resize window");
+  }
+}
+
+function loadGrid(start: number) {
+  startIndex = start;
+  endIndex = start + rowsPerView;
+
+  if (startIndex < 0) {
+    startIndex = 0;
+    endIndex = rowsPerView;
+  }
+  if (endIndex >= recordCount) {
+    endIndex = recordCount - 1;
+  }
+  if (startIndex >= recordCount) {
+    startIndex = recordCount - rowsPerView;
+  }
+  if (startIndex >= recordCount - rowsPerView && startIndex <= recordCount - 1) {
+    startIndex = recordCount - (rowsPerView + 1);
+  }
+  // calling Records
+  getRecords(startIndex, endIndex);
+}
+
+ // calling record count from the api, full url http://localhost:2050/recordcount
+function getRecordCount() {
+  $.get("/recordCount", function (numberRecords) {
+    recordCount = numberRecords;
+    getColumnHeaders();
+  }).fail(function () {
+    alert("Error while calling Record Count");
+  });
+}
+
+// calling column header from the api, full url http://localhost:2050/columns
+function getColumnHeaders() {
+  $.get("/columns", function (headers) {
+    columnHeaders = JSON.parse(headers);
+    loadGrid(0);
+  }).fail(function () {
+    alert("Error while calling column header");
+  });
+}
+
+// calling records from the api, full url http://localhost:2050/records?from=1&to=3
+function getRecords(start: number, end: number) {
+  $.get("/records?from=" + start + "&to=" + end, function (rec) {
+    //creating table grid with records returned by the api
+    createTable(JSON.parse(rec));
+  }).fail(function () {
+    alert("Couldn't load records , page will refresh");
+  });
+}
+
+// building table
+function createTable(records: any[]) {
+ // clearing current table
+  document.getElementById("divTable").innerHTML = "";
```

Comment:
I recommend you only clean out the table the moment before you set the new one. That will avoid any annoying flashes on slow machines. (Move this line to just before 102)